### DISCUSSION TITLE
Cannot find module use-sync-external-store

### DISCUSSION THREAD

**Discussion by nkleidis** [pmndrs/zustand#2853] [General] [answered]
Hello,
I am trying to upgrade zustand to version 5.0.1 but I get the following error when I am trying to run the project or the tests:

```
Cannot find module 'use-sync-external-store/shim/with-selector' from 'node_modules/zustand/traditional.js'

    Require stack:
      node_modules/zustand/traditional.js
     ....
```

After manually installing `npm i use-sync-external-store` the error goes away. (The tests take too long to complete but I guess this is the thing with the new version and we need to fix the bad practices in our app)

Has anyone had this issue?

I am using:
```
"react": "18.3.1",
"react-native": "0.75.4",
"zustand": "5.0.1",
"typescript": "5.6.3"
```

**Comment by dai-shi** [comment]
> After manually installing `npm i use-sync-external-store` the error goes away.

That's the correct solution. If you find a way to improve docs, feel free to send a PR.

**Reply by nkleidis** [reply]
Thank @dai-shi for your answer! I somehow missed this on the migration guide. 
I believe though that it could be mentioned as well in the docs apart from the migration guide.
It would be beneficial for people using this library for the first time (meaning they don't need to go through the migration guide)
What do you think?

**Reply by dai-shi** [reply, marked as answer]
Yeah, maybe there should be a note about it. Like this one?

https://github.com/pmndrs/zustand/blob/5a8842fbc3bda64fa9e59111e49697caff6fa64c/docs/apis/create-with-equality-fn.md?plain=1#L63

I wonder if @dbritto-dev has any idea.

It's intentional that we don't talk about zustand/traditional in readme.md, so this should be somewhere else in the docs.

**Reply by dbritto-dev** [reply]
@dai-shi sure, let me add it

**Comment by khuddite** [comment]
Thanks! `npm i use-sync-external-store` fixed the issue for me :)

Btw, it started happening after upgrading React v18 to v19 and Next v14 to v15. I am not sure how the upgrades are related to the issue, though.

**Comment by dbritto-dev** [comment]
I updated the docs to inform that -> https://zustand.docs.pmnd.rs/apis/create-with-equality-fn and https://zustand.docs.pmnd.rs/hooks/use-store-with-equality-fn

@dai-shi  FYI
